### DISCUSSION TITLE
Add spacing from Input to keyboard

### DISCUSSION THREAD

**Discussion by tanlucvo** [mateusz1913/react-native-avoid-softinput#179] [Q&A]
### Feature request

How to add spacing , In this case I must scroll bottom to show content below

https://github.com/mateusz1913/react-native-avoid-softinput/assets/57494001/a4f61ae7-53aa-4b20-8135-19e0ab5e04aa

**Comment by mateusz1913** [comment]
Use [`AvoidSoftInput.setAvoidOffset`](https://mateusz1913.github.io/react-native-avoid-softinput/docs/api/module/set-avoid-offset) or [`avoidOffset`](https://mateusz1913.github.io/react-native-avoid-softinput/docs/api/view/#props) depending on what API you're using in this case

**Reply by tanlucvo** [reply]
It delay and create white space below input, seem that offset for parent


https://github.com/mateusz1913/react-native-avoid-softinput/assets/57494001/a30a9522-619a-47a0-a2e2-0880131e734e



<img width="396" alt="258645054-0df59dcd-5c14-49bc-a9a6-d517cb876f06" src="https://github.com/mateusz1913/react-native-avoid-softinput/assets/57494001/f422c4ed-8379-41db-be5b-85d68bd3df21">

**Reply by bryanprimus** [reply]
Instead of setting the offset, you can probably just apply a `paddingBottom` to your ChatBox container.

**Reply by tanlucvo** [reply]
I tried but i got same result, i found this lib react-native-keyboard-manager has solution for this issue. But I still want to use this component ^-^

**Reply by mateusz1913** [reply]
@tanlucvo could you elaborate more on what spacing you actually want to add? The initial description of your case is unclear to me

**Reply by tanlucvo** [reply]
In this video, i wrap input in View have backgroundColor blue. 

https://github.com/mateusz1913/react-native-avoid-softinput/assets/57494001/70fb9718-63f9-4a06-8a88-b6bc7917375d

after keyboard show:
![Simulator Screen Shot - iPhone 14 Pro - 2023-08-07 at 10 47 13](https://github.com/mateusz1913/react-native-avoid-softinput/assets/57494001/b64b796d-8920-4912-b72a-92db317d442e)
But i want all view backgroundColor blue above keyboard, I don't need to scroll bottom. picture below
![Simulator Screen Shot - iPhone 14 Pro - 2023-08-07 at 10 47 19](https://github.com/mateusz1913/react-native-avoid-softinput/assets/57494001/84b4e2ea-ae1b-497c-a305-1bcf0f33368b)

**Comment by mateusz1913** [comment]
@tanlucvo the padding added by library is added to the input element and not the arbitrary container that wraps the input. You need to  rearrange the styles in your case, so that the blue container has no padding at the bottom and the input has some padding bottom provided by you (+ the one added by library when the keyboard is displayed)

**Comment by traxx10** [comment]
I have the same issue.
In my case, i don't want padding but i want the input to scroll a bit more to the top so the contents below the input can show.

For example, i have an address input but as the user types, they can't see the suggestions below unless they scroll. 

https://github.com/mateusz1913/react-native-avoid-softinput/assets/31565764/b99c425d-5a9f-4f1d-b4d3-44fb7dfccbdb

**Reply by mateusz1913** [reply]
Hi @traxx10, like I explained before - there's no way for the library to know which UI elements visually belong to your custom text field component (e.g. some info/error labels displayed below the text field).
The library only detects the text field native element (UITextField/UITextView on iOS and EditText on Android) and tries to make those exact elements visible.

It's not possible for any library, to predict where some custom text field component (that's wrapped with some style container and additional elements like labels) begin or end its bounds on the screen. The best it can do is to locate the field element with which user will be interacting.

**Reply by traxx10** [reply]
Okay thank you

**Comment by GollyJer** [comment]
Hi @mateusz1913. I came to post about this same issue and found this discussion.
Does `AvoidSoftInput` have any knowledge of the scroll position after the keyboard is fully shown?

I was hoping to tackle this with something like...
```javascript
useSoftInputShown(({ scrollPositionY }) => {
    scrollViewRef.current?.scrollTo({
      y: scrollPositionY + 20,
      animated: true,
    });
  });
```

I realize `scrollPositionY` doesn't exist. But wondering if possible?
The other thing I realized while trying to figure something out is that `useSoftInputShown` actually fires before the keyboard is shown.
It's more like `useSoftInputWillShow`. I was trying to store the scroll position separately and update a local piece of state with the Y position via `onScroll` when I realized there was no way to capture the end of the keyboard scrolling. `onScrollMomentumEnd` isn't triggered by AvoidSoftInput.

Basically what I'm trying to accomplish is
 - wait for keyboard to fully show.
 - get the current y position
 - move it just a little more

Thanks for any ideas!

**Reply by mateusz1913** [reply]
Hi @GollyJer, I believe it should just be possible to assign `onScroll` listener to the `ScrollView` you want to scroll and when the keyboard is displayed and the `onScroll` events fired, you can somehow trigger `scrollTo` method.
I don't plan to add any `scrollPositionY` value, as it's not the responsibility of this library to retrieve such value + it would be super complex to achieve that
